requests/adapters.py

```diff
@@ -300,6 +301,10 @@ def get_connection(self, url, proxies=None):
 
         if proxy:
             proxy = prepend_scheme_if_needed(proxy, 'http')
+            proxy_url = parse_url(proxy)
+            if not proxy_url.host:
+                raise ProxyError("Please check proxy URL. It is malformed "
+                                 "and could be missing the host")
```

Comment:
Could you end this last sentence with a period?

Reply:
Sure, done.